nerdtree_plugin/git_status.vim

```diff
@@ -73,7 +93,30 @@ function! g:NERDTreeGitStatusRefresh()
     let b:NOT_A_GIT_REPOSITORY        = 1
 
     let l:root = b:NERDTree.root.path.str()
-    let l:gitcmd = 'git -c color.status=false status -s'
+    " them     us      Meaning
+    " -------------------------------------------------
+    "           [MD]   not updated
+    " M        [ MD]   updated in index
+    " A        [ MD]   added to index
+    " D         [ M]   deleted from index
+    " R        [ MD]   renamed in index
+    " C        [ MD]   copied in index
+    " [MARC]           index and work tree matches
+    " [ MARC]     M    work tree changed since index
+    " [ MARC]     D    deleted in work tree
+    " -------------------------------------------------
+    " D           D    unmerged, both deleted
+    " A           U    unmerged, added by us
+    " U           D    unmerged, deleted by them
+    " U           A    unmerged, added by them
+    " D           U    unmerged, deleted by us
+    " A           A    unmerged, both added
+    " U           U    unmerged, both modified
+    " -------------------------------------------------
+    " ?           ?    untracked
+    " !           !    ignored
+    " -------------------------------------------------
+    let l:gitcmd = 'git status --porcelain -uall | sed -E ''s/^UU|^UD|^UA|^DU|^DD|^DA|^AU|^AA/1/g'' | sed -E ''s/^MM|^M |^ M|^  /2/g'' | sed -E ''s/^A\?|^A |^AM|^\?\?|^\? |^\?M/3/g'' | sed -E ''s/^RM|^R /4/g'' | sed -E ''s/^D |^DM/5/g'' | sort'
```

Comment:
Have you tested under a dir contains very very modified or other status files?  Maybe this huge pipeline will make the loading speed be slow.